packages/dart_frog_auth/lib/src/dart_frog_auth.dart

```diff
@@ -90,21 +105,37 @@ Middleware basicAuthentication<T extends Object>({
 /// [RequestContext]. If the function returns false, the middleware will not
 /// apply to the route and the call will have no authentication validation.
 Middleware bearerAuthentication<T extends Object>({
-  required Future<T?> Function(String) userFromToken,
+  Future<T?> Function(String)? userFromToken,
+  Future<T?> Function(String, RequestContext)? userFromContext,
```

Comment:
**Nit:**

I would provide the names of the arguments so when implemented they are kept consistent. Same applies to others.
```suggestion
  Future<T?> Function(String token, RequestContext context)? userFromContext,
```